cpp/src/arrow/ipc/metadata_internal.cc

```diff
@@ -957,6 +957,67 @@ Status MakeSparseMatrixIndexCSX(FBB& fbb, const SparseIndexType& sparse_index,
   return Status::OK();
 }
 
+Status MakeSparseTensorIndexCSF(FBB& fbb, const SparseCSFIndex& sparse_index,
+                                const std::vector<BufferMetadata>& buffers,
+                                flatbuf::SparseTensorIndex* fb_sparse_index_type,
+                                Offset* fb_sparse_index, size_t* num_buffers) {
+  *fb_sparse_index_type = flatbuf::SparseTensorIndex::SparseTensorIndexCSF;
+  const int ndim = static_cast<int>(sparse_index.axis_order().size());
+
+  // We assume that the value type of indptr tensor is an integer.
+  const auto& indptr_value_type =
+      checked_cast<const IntegerType&>(*sparse_index.indptr()[0]->type());
+  auto indptr_type_offset = flatbuf::CreateInt(fbb, indptr_value_type.bit_width(),
+                                               indptr_value_type.is_signed());
+
+  // We assume that the value type of indices tensor is an integer.
+  const auto& indices_value_type =
+      checked_cast<const IntegerType&>(*sparse_index.indices()[0]->type());
+  auto indices_type_offset = flatbuf::CreateInt(fbb, indices_value_type.bit_width(),
+                                                indices_value_type.is_signed());
+
+  const int64_t indptr_elem_size = indptr_value_type.bit_width() / 8;
+  const int64_t indices_elem_size = indices_value_type.bit_width() / 8;
+
+  int64_t indptr_offset = 0;
+  int64_t indices_offset = 0;
+  std::vector<flatbuf::Buffer> indptr, indices;
+
+  for (const std::shared_ptr<arrow::Tensor> tensor : sparse_index.indptr()) {
+    const int64_t size = tensor->data()->size() / indptr_elem_size;
+    const int64_t padding = BitUtil::RoundUpToMultipleOf8(size * indptr_elem_size) - size;
+
+    indptr.push_back({indptr_offset, size});
+    indptr_offset += size + padding;
+  }
+  indices_offset = indptr_offset;
+  for (const std::shared_ptr<arrow::Tensor> tensor : sparse_index.indices()) {
+    const int64_t size = tensor->data()->size() / indices_elem_size;
+    const int64_t padding =
+        BitUtil::RoundUpToMultipleOf8(size * indices_elem_size) - size;
+
+    indices.push_back({indices_offset, size});
+    indices_offset += size + padding;
+  }
+
+  auto fb_indices = fbb.CreateVectorOfStructs(indices);
+  auto fb_indptr = fbb.CreateVectorOfStructs(indptr);
+
+  std::vector<int> axis_order;
+  for (int i = 0; i < ndim; ++i) {
+    axis_order.emplace_back(static_cast<const int>(sparse_index.axis_order()[i]));
```

Comment:
Why you cast the value to `const int` instead of just `int`?

Reply:
Changed.